fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java

```diff
@@ -348,17 +353,45 @@ public CompletableFuture<Boolean> markIndexReadable(boolean markReadablePlease)
         if (!markReadablePlease) {
             return AsyncUtil.READY_FALSE; // they didn't say please..
         }
-        AtomicBoolean allReadable = new AtomicBoolean(true);
-        return getRunner().runAsync(context -> openRecordStore(context)
-                .thenCompose(store -> forEachTargetIndex(index ->
-                        store.markIndexReadable(index).thenApply(built -> {
-                            if (!built) {
-                                allReadable.set(false);
-                            }
-                            return null;
-                        }))
-                        .thenApply(ignore -> allReadable.get())
-                ), common.indexLogMessageKeyValues("IndexingBase::markReadable"));
+
+        AtomicReference<RecordIndexUniquenessViolation> uniquenessViolationExRef = new AtomicReference<>();
+        AtomicBoolean allReadable = new AtomicBoolean();
+
+        // Mark each index readable in its own (retriable, parallel) transaction
+        return forEachTargetIndex(index ->
+                markIndexReadableSingleTarget(index, allReadable, uniquenessViolationExRef)
+        ).thenCompose(ignore -> {
+            RecordIndexUniquenessViolation ex = uniquenessViolationExRef.get();
+            if (ex != null) {
+                throw ex;
+            }
+            return allReadable.get() ? AsyncUtil.READY_TRUE : AsyncUtil.READY_FALSE;
+        });
+    }
+
+    private CompletableFuture<Boolean> markIndexReadableSingleTarget(Index index, AtomicBoolean allReadable,
+                                                                     AtomicReference<RecordIndexUniquenessViolation> uniquenessViolationExRef) {
+        // An extension function to reduce markIndexReadable's complexity
+        return common.getRunner().runAsync(context ->
```

Comment:
In addition to potentially changing the state to `UNIQUE_PENDING`, this changes it so that if doing a multi-index build, and one index cannot be marked readable, the others can be right? I don't think that's clear from the issue or release notes, so probably worth calling out in one of those places.